AINFRA-780 - Make NetworkingTests a Swift package test target

mokagio opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Now that all the frameworks sources are Swift packages (targets in the same Swift package to be precise) we can move the tests there, too. We didn't do that all in one go to avoid big diffs.

Notice:

- The whole `Networking/` folder has been removed, including the dedicated Xcode project
- No changes in the test code
- The Yosemite and WooCommerce test targets now reference the `NetworkingTests/Responses/` folder from the location in the `Modules/Tests/` folder. At some point, we ought to isolate the responses in some kind of tests helper package, but for the moment including them this way (which to me is like a soft-duplication) is convenient.

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
See green CI.

Additionally, you can run the focused Hardware tests by selecting the Hardware scheme from Xcode.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. — N.A.

mokagio committed:
AINFRA-779 - Make NetworkingTests a Swift package test target

dangermattic commented:
<!--
  0 Errors
  2 Warnings: <code>Modules/Package.swift</c...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

jaclync commented on Modules/Package.swift:
```diff
@@ -228,6 +228,19 @@ let package = Package(
+                "WordPressShared",
+                .product(name: "KeychainAccess", package: "KeychainAccess"),
+            ],
+            resources: [.process("Responses")]
```
> > * The Yosemite and WooCommerce test targets now reference the `NetworkingTests/Responses/` folder from the location in the `Modules/Tests/` folder. At some point, we ought to isolate the responses in some kind of tests helper package, but for the moment including them this way (which to me is like a soft-duplication) is convenient.

I agree that a separate package for these mock responses would be ideal.

jaclync reviewed (commented):
Thanks for the changes! I removed the derived data and tried running the remote tests locally, some behaviors I'd like to clarify:

1. There's still a Networking project in the project navigator, but clicking on shows an error (first screenshot). Also seeing an error (the second screenshot) that doesn't block running the tests but I think it's worth removing to reduce confusion:

error clicking on Networking project | similar error
-- | --
<img width="1624" alt="Screenshot 2025-06-23 at 7 44 15 AM" src="https://github.com/user-attachments/assets/5020e186-c6a0-4832-946b-b74e1ea83aa5" /> | <img width="482" alt="Screenshot 2025-06-23 at 7 46 31 AM" src="https://github.com/user-attachments/assets/0f117f86-8a01-47bb-8ca0-fd3670838058" />

2. In order to run Networking tests, is it an expected change to have to switch to the Networking scheme instead of staying in the WooCommerce scheme?

<img width="1624" alt="Screenshot 2025-06-23 at 7 44 15 AM" src="https://github.com/user-attachments/assets/7810d27a-d495-485c-8192-407647a857a9" />

3. I haven't been able to see the diamond check boxes next to each test class/function in the code to run them, do they show up on your end? Just wondering if it is an unexpected change, if so it might be worth notifying developers to prevent from unncessary time trying to get them working. I tried reopening Xcode which usually resolves it, but still cannot see them appear.

branch | trunk
-- | --
<img width="1624" alt="Screenshot 2025-06-23 at 7 51 02 AM" src="https://github.com/user-attachments/assets/54810e7d-874d-4942-9d77-f25b96828184" /> |  <img width="970" alt="Screenshot 2025-06-23 at 7 54 24 AM" src="https://github.com/user-attachments/assets/01ebe858-f2f9-43c9-82a2-ec0c2adeeda4" />

mokagio commented:
Thanks for the review @jaclync 🙇‍♂️ 

> 1. There's still a Networking project in the project navigator...

Ouch! I should have double checked and removed it from the workspace. I rebased on `trunk` to address a merge conflict and also addressed that.

<img width="858" alt="image" src="https://github.com/user-attachments/assets/c9b5104e-a6a7-4c58-af38-c22e7eaa9dcb" />

> 2. In order to run Networking tests, is it an expected change to have to switch to the Networking scheme instead of staying in the WooCommerce scheme?

No. That's just Xcode being annoying with the test plan and rebase. See for example https://github.com/woocommerce/woocommerce-ios/pull/15794/files#r2160527954. This should be addressed now as well.

<img width="1189" alt="image" src="https://github.com/user-attachments/assets/2e5579fa-da5b-4f8a-8902-0afb4b75254e" />

> 3. I haven't been able to see the diamond check boxes next to each test class/function in the code to run them, do they show up on your end?

Funny that you mention that. It's a long standing Xcode bug which [_should_ be fixed in Xcode 26:](https://x.com/mokagio/status/1932311096154808342)

![image](https://github.com/user-attachments/assets/e41fe6a5-5933-4157-88a5-42e411c99467)

jaclync commented:
Thanks for the updates! 1 is fixed now, and 3 doesn't sound like a blocking issue and should be fixed later.

> > 2\. In order to run Networking tests, is it an expected change to have to switch to the Networking scheme instead of staying in the WooCommerce scheme?
> 
> No. That's just Xcode being annoying with the test plan and rebase. See for example https://github.com/woocommerce/woocommerce-ios/pull/15794/files#r2160527954. This should be addressed now as well.

After rebasing the PR and relaunching Xcode a few times, I still wasn't able to run the NetworkingTests with the WooCommerce scheme.

<img width="968" alt="Screenshot 2025-06-24 at 6 00 42 AM" src="https://github.com/user-attachments/assets/699c74fa-aa27-4a13-862e-62334ab660e5" />

More importantly, I didn't seem to find the NetworkingTests (and StorageTests which is also missing from the screenshot above) mentions in the [unit tests CI logs](https://buildkite.com/automattic/woocommerce-ios/builds/30673#01979f43-49ca-44bb-898b-a69a4bae1f07). Not sure if I missed the tests in the CI logs?

mokagio committed:
Restore Networking and Storage tests in UnitTests.xctestplan once more

See:

- https://github.com/woocommerce/woocommerce-ios/pull/15796#issuecomment-2998337323
- https://github.com/woocommerce/woocommerce-ios/pull/15796#issuecomment-2999643970

mokagio commented:
@jaclync  Mmmm.... What's going on? The two targets weren't there on my branch after rebasing on `trunk` either. But that's very odd given the screenshot above, which I took on purpose to show they worked when I pushed.

Anyway, I committed the Test Plan update in isolation in 93ec55a96b. 

Let's see how it goes in CI this time.

iamgabrielma commented:
👀  Just sneaking in this PR to share that CI failed with the crash I [originally saw](https://github.com/woocommerce/woocommerce-ios/pull/15794#pullrequestreview-2948560402) when making StorageTests a package, so we might have some flakiness here:

```
🔬 Unit Tests: 1 tests have failed (1 distinct assertion failure(s) in total)
test_file_is_loaded() in FileStorageTests
Crash: xctest (2683) implicit closure #1 in FileStorageTests.test_file_is_loaded()
```

I haven't re-run the failing build in case you want to take a further look 👍

mokagio committed:
Fix `Bundle` access in storage tests

mokagio commented:
@jaclync @iamgabrielma thanks folks.

Everything should be fixed now. I clearly lost something along the way with the rebases 😓

jaclync reviewed (approved):
Thanks for the updates @mokagio! I see the NetworkingTests in the latest CI log, can run NetworkingTests without switching scheme as the NetworkingTests is included in the test plan now. :shipit: 

<img width="1624" alt="Screenshot 2025-06-25 at 8 04 14 AM" src="https://github.com/user-attachments/assets/8b69245b-5625-47e1-b5ea-9b7be8dd7255" />
